pkg/integrationtests/integrationtests.go

```diff
@@ -0,0 +1,233 @@
+					Name:     "no-ident-root-dir",
+				},
+			},
+			"license": {
+				{
+					Severity: "error",
+					Title:    "LICENSE file could not be parsed.",
+					Detail:   "Could not parse the license file inside the plugin archive. Please make sure to include a valid license in your LICENSE file in your archive.",
+					Name:     "license-not-provided",
+				},
+			},
+			"manifest": {
+				{
+					Severity: "warning",
+					Title:    "unsigned plugin",
+					Detail:   "This is a new (unpublished) plugin. This is expected during the initial review process. Please allow the review to continue, and a member of our team will inform you when your plugin can be signed.",
+					Name:     "unsigned-plugin",
+				},
+			},
+			"metadatapaths": {
+				{
+					Severity: "error",
+					Title:    "plugin.json: small logo path doesn't exists: img/logo.svg",
+					Detail:   "Refer only existing files. Make sure the files referred in plugin.json are included in the archive.",
+					Name:     "path-not-exists",
+				},
+				{
+					Severity: "error",
+					Title:    "plugin.json: large logo path doesn't exists: img/logo.svg",
+					Detail:   "Refer only existing files. Make sure the files referred in plugin.json are included in the archive.",
+					Name:     "path-not-exists",
+				},
+			},
+			"metadatavalid": {
+				{
+					Severity: "warning",
+					Title:    "plugin.json: dependencies: grafanaDependency is required",
+					Detail:   "The plugin.json file is not following the schema. Please refer to the documentation for more information. https://grafana.com/docs/grafana/latest/developers/plugins/metadata/",
+					Name:     "invalid-metadata",
+				},
+			},
+			"readme": {
+				{
+					Severity: "error",
+					Title:    "README.md is empty",
+					Detail:   "A README.md file is required for plugins. The contents of the file will be displayed in the Plugin catalog.",
+					Name:     "missing-readme",
+				},
+			},
+			"screenshots": {
+				{
+					Severity: "warning",
+					Title:    "plugin.json: should include screenshots for the Plugin catalog",
+					Detail:   "Screenshots are displayed in the Plugin catalog. Please add at least one screenshot to your plugin.json.",
+					Name:     "screenshots",
+				},
+			},
+		},
+	},
+}
+
+func RunTests(binary string) error {
+	env := []string{
+		"DEBUG=0",
+		"OPENAI_API_KEY=",
+	}
+
+	basePath := "./testdata"
+
+	hasError := false
+	finalOutput := ""
+
+	for file := range files {
+		fmt.Printf("Running %s\n", file)
+		command := fmt.Sprintf(
+			"%s -config ./config/integration-tests.yaml %s",
+			binary,
+			filepath.Join(basePath, file),
+		)
+		fmt.Printf("Running command: %s\n", command)
+		cmd := exec.Command("sh", "-c", command)
+		var outb, errb bytes.Buffer
+		cmd.Stdout = &outb
+		cmd.Stderr = &errb
+		cmd.Env = append(os.Environ(), env...)
+		err := cmd.Run()
+		if err != nil && len(outb.String()) == 0 {
```

Comment:
Nit: I would probably also assert on the exit code of the validator. For the `invalid` zips in particular, the exit code is 1.

To do this, I would add a new `TestCase` struct:

```go
type TestCase struct {
	expExitCode int
	expReport   JsonReport
}
```

and change the `files` map so it's a `map[string]TestCase{`

and for the invaid ones:

```go
	"invalid.zip": {
		expExitCode: 1,
		expReport: JsonReport{
			Id:      "invalid-panel",
			Version: "1.0.0",
			PluginValidator: map[string][]Issue{
				"archive": {
					{
						Severity: "error",
						Title:    "Archive contains more than one directory",
						Detail:   "Archive should contain only one directory named after plugin id. Found 7 directories",
						Name:     "more-than-one-dir",
					},
					{
						Severity: "error",
						Title:    "Plugin archive is improperly structured",
						Detail:   "",
						Name:     "zip-invalid",
					},
				},
			},
		},
	},
```

Then, to assert it:

```go
err := cmd.Run()
if tc.expExitCode == 0 {
	require.NoError(t, err)
} else {
	require.Equal(t, tc.expExitCode, cmd.ProcessState.ExitCode())
	require.Error(t, err)
}
require.NotEmpty(t, outb)
```